servers/zms/src/main/java/com/yahoo/athenz/zms/store/impl/jdbc/JDBCConnection.java

```diff
@@ -2462,7 +2474,10 @@ boolean insertPendingRoleMember(int roleId, int principalId, RoleMember roleMemb
 
         int affectedRows;
         if (roleMemberExists) {
-
+            // check first that the current request doesn't contradict the existing one
+            if (conflictRoleMemberPendingRequest(roleId, roleMember.getMemberName(), roleMember.getPendingState())) {
```

Comment:
this actually generates another mysql call which I believe is unnecessary. before calling this method we already take a hit and call roleMemberExists method which checks if the member already exists or not. we can just update that query to return the state as well and if the request is a pending one we can check if the match and return exception there.